contracts/cw20-merkle-airdrop/src/contract.rs

```diff
@@ -0,0 +1,486 @@
+    let latest_stage = STAGE.load(deps.storage)?;
+    let resp = LatestStageResponse { latest_stage };
+
+    Ok(resp)
+}
+
+pub fn query_is_claimed(deps: Deps, stage: u8, address: String) -> StdResult<IsClaimedResponse> {
+    let key: (&Addr, U8Key) = (&deps.api.addr_validate(&address)?, stage.into());
+    let is_claimed = CLAIM.load(deps.storage, key)?;
+    let resp = IsClaimedResponse { is_claimed };
+
+    Ok(resp)
+}
+
+#[entry_point]
+pub fn migrate(_deps: DepsMut, _env: Env, _msg: MigrateMsg) -> StdResult<Response> {
+    Ok(Response::default())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info};
+    use cosmwasm_std::{from_binary, CosmosMsg};
+
+    #[test]
+    fn proper_instantiation() {
+        let mut deps = mock_dependencies(&[]);
+
+        let msg = InstantiateMsg {
+            owner: "owner0000".to_string(),
+            cw20_token_address: "anchor0000".to_string(),
+        };
+
+        let env = mock_env();
+        let info = mock_info("addr0000", &[]);
+
+        // we can just call .unwrap() to assert this was a success
+        let _res = instantiate(deps.as_mut(), env.clone(), info, msg).unwrap();
+
+        // it worked, let's query the state
+        let res = query(deps.as_ref(), env.clone(), QueryMsg::Config {}).unwrap();
+        let config: ConfigResponse = from_binary(&res).unwrap();
+        assert_eq!("owner0000", config.owner.as_str());
+        assert_eq!("anchor0000", config.cw20_token_address.as_str());
+
+        let res = query(deps.as_ref(), env, QueryMsg::LatestStage {}).unwrap();
+        let latest_stage: LatestStageResponse = from_binary(&res).unwrap();
+        assert_eq!(0u8, latest_stage.latest_stage);
+    }
+
+    #[test]
+    fn update_config() {
+        let mut deps = mock_dependencies(&[]);
+
+        let msg = InstantiateMsg {
+            owner: "owner0000".to_string(),
+            cw20_token_address: "anchor0000".to_string(),
+        };
+
+        let env = mock_env();
+        let info = mock_info("addr0000", &[]);
+        let _res = instantiate(deps.as_mut(), env, info, msg).unwrap();
+
+        // update owner
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::UpdateConfig {
+            owner: Some("owner0001".to_string()),
+        };
+
+        let res = execute(deps.as_mut(), env.clone(), info, msg).unwrap();
+        assert_eq!(0, res.messages.len());
+
+        // it worked, let's query the state
+        let res = query(deps.as_ref(), env, QueryMsg::Config {}).unwrap();
+        let config: ConfigResponse = from_binary(&res).unwrap();
+        assert_eq!("owner0001", config.owner.as_str());
+
+        // Unauthorized err
+        let env = mock_env();
+        let info = mock_info("owner0000", &[]);
+        let msg = ExecuteMsg::UpdateConfig { owner: None };
+
+        let res = execute(deps.as_mut(), env, info, msg);
+        match res {
```

Comment:
Modern rust allows `matches!` for this.
Also, you may be able to use PartialEq.

```rust
let err = execute(deps.as_mut(), env, info, msg).unwrap_err();
assert_eq!(err, ContractError::Unauthorized{});
```